spec/21-http_response_header_format.md

```diff
@@ -0,0 +1,197 @@
+# Trace Context HTTP Response Headers Format
+
+This section describes the binding of the distributed trace context to the `traceresponse` HTTP header.
+
+## Traceresponse Header
+
+The `traceresponse` HTTP response header field identifies a completed request in a tracing system. It has four fields:
+
+* `version`
+* `trace-id`
+* `parent-id`
+* `trace-flags`
+
+### Header Name
+
+Header name: `traceresponse`
+
+In order to increase interoperability across multiple protocols and encourage successful integration, the header name SHOULD be lowercase. The header name is a single word without any delimiters, for example, a hyphen (`-`).
+
+Vendors MUST expect the header name in any case (upper, lower, mixed), and SHOULD send the header name in lowercase.
+
+### traceresponse Header Field Values
+
+
+This section uses the Augmented Backus-Naur Form (ABNF) notation of [[!RFC5234]], including the DIGIT rule from that document. The `DIGIT` rule defines a single number character `0`-`9`.
+
+``` abnf
+HEXDIGLC = DIGIT / "a" / "b" / "c" / "d" / "e" / "f" ; lowercase hex character
+value           = version "-" version-format
+```
+
+The dash (`-`) character is used as a delimiter between fields.
+
+#### version
+
+``` abnf
+version         = 2HEXDIGLC   ; this document assumes version 00. Version 255 is forbidden
+```
+
+The value is US-ASCII encoded (which is UTF-8 compliant).
+
+Version (`version`) is 1 byte representing an 8-bit unsigned integer. Version `255` is invalid. The current specification assumes the `version` is set to `00`.
+
+#### version-format
+
+The following `version-format` definition is used for version `00`.
+
+``` abnf
+version-format   = [trace-id] "-" [parent-id] "-" [trace-flags]
```

Comment:
Making fields optional while keeping the same format as `traceparent` carry potential issues. Some may decide to use the same optionality for `traceparent` (even though it's not allowed by spec) and break interoperability. Mostly because the same code may be used to parse and generate both headers.

This said, it will be great to confirm that all the scenario listed below in this proposal:
1. Would actually significantly benefit from fields optionality.
2. Define a clear logic on how to decide which parts to return.
3. Would not need more properties than listed in `traceresponse` and will ACTUALLY need the arbitrary `tracestate` on response instead.